### ISSUE TITLE
More list/watchlist methods

### ISSUE THREAD

**Issue by Razzeee** [fuzeman/trakt.py#26]
So there is an watchlist method, but nothing to add to it?
Also what about getting all custom lists and getting the items of them?

**Comment by fuzeman** [comment]
Watchlists should be properly supported now, working on custom lists now.

-----------------------------------------

#### Watchlist Examples

**Retrieve items**
```python
# Retrieve shows
shows = {}

Trakt['sync/watchlist'].shows(shows)
Trakt['sync/watchlist'].seasons(shows)
Trakt['sync/watchlist'].episodes(shows)

# Retrieve movies
movies = Trakt['sync/watchlist'].movies()
```

**Add items to watchlist**
```python
Trakt['sync/watchlist'].add({
    'movies': [
        {'ids': {'tmdb': 135397}}
    ],
    'shows': [
        {'ids': {'tvdb': 73255}},
        {'ids': {'tvdb': 80348}, 'seasons': [{'number': 1, 'episodes': [{'number': 1}]}]}
    ]
})
```

**Remove items from watchlist**
```python
Trakt['sync/watchlist'].remove({
    'shows': [
        {'ids': {'tvdb': 73255}}
    ]
})
```

**Comment by fuzeman** [comment]
I've nearly finished custom lists support, just working on some tests for it before merging into the develop branch.

Let me know if you have any questions about it.

---------------------------------

### Custom List

#### Examples

**Retrieve all lists for a user**
```python
custom_list = Trakt['users/me/lists'].get()
```

**Retrieve a single list**
```python
custom_list = Trakt['users/me/lists/*'].get(l.id)
```

**Retrieve items for a list**
```python
items = Trakt['users/me/lists/*'].items(l.id)
```

**Create a new list**
```python
custom_list = Trakt['users/me/lists'].create(name='Movies')
```

**Add items to a list**
```python
custom_list.add({
    'movies': [
        {'ids': {'tmdb': '76341'}},
        {'ids': {'tmdb': '257091'}},
        {'ids': {'tmdb': '287424'}}
    ]
})
```

**Remove items from a list**
```python
custom_list.remove({
    'movies': [
        {'ids': {'tmdb': '257091'}}
    ]
})
```

**Update a list**
```python
custom_list.update(name="My Movies")
```

#### Methods

**User Lists interface**
```python
Trakt['users/*/lists'].create(username, name, description=None, privacy='private', display_numbers=False, allow_comments=True)
Trakt['users/*/lists'].get(username)

# Note: username can be provided in the path, for example:
Trakt['users/me/lists'].get()
```

**User List interface**
```python
Trakt['users/*/lists/*'].get(username, id)
Trakt['users/*/lists/*'].items(username, id)

Trakt['users/*/lists/*'].add(username, id, items)
Trakt['users/*/lists/*'].delete(username, id)
Trakt['users/*/lists/*'].update(username, id, name=None, description=None, privacy=None, display_numbers=None, allow_comments=None)
Trakt['users/*/lists/*'].remove(username, id, items)
Trakt['users/*/lists/*'].like(username, id)
Trakt['users/*/lists/*'].unlike(username, id)

# Note: username and/or list id can be provided in the path, for example:
Trakt['users/me/lists/movies'].get()
```

**CustomList object**
```python
CustomList.add(items)
CustomList.delete()
CustomList.update(name=None, description=None, privacy=None, display_numbers=None, allow_comments=None)
CustomList.remove(items)
CustomList.like()
CustomList.unlike()
```

**Comment by Razzeee** [comment]
Looks good, I will figure out my understanding problems once you finished it and I had a go at it :)

**Comment by fuzeman** [comment]
Just merged the finished custom lists support into the develop branch, let me know if you find any bugs.

**Comment by Razzeee** [comment]
Watchlist is working fine so far

**Comment by fuzeman** [comment]
Released in [v2.4.0](https://github.com/fuzeman/trakt.py/releases/tag/2.4.0)
